### PR TITLE
Re-work publish workflow to automatically set version if not defined

### PR DESCRIPTION
**Pull request by Hinton**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13058

## 📔 Objective

Publish wasm automatically on merge to main. Determine version from job-id and branch.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by Hinton** [comment]
@bitwarden/dept-bre is there a way to ensure we only publish wasm IF the "run tests" workflow succeeds? I.e. depend on the result of another workflow.

File: .github/workflows/build-wasm-internal.yml [hunk 1]
```diff
@@ -73,3 +73,33 @@ jobs:
...
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          github-token: ${{ steps.retrieve-secret-pat.outputs.github-pat-bitwarden-devops-bot-repo-scope }}
+          script: |
+            await github.rest.actions.createWorkflowDispatch({
+              owner: 'bitwarden',
+              repo: 'sdk-internal',
+              workflow_id: 'publish-wasm-internal.yml',
+              ref: 'main',
+            })
```
Comment (coroiu) [inline .github/workflows/build-wasm-internal.yml:112]: question: is this how GH actions work? You have to trigger another workflow using a script?

**Review by coroiu — DISMISSED** [review]

**Comment by vgrassia** [comment]
> @bitwarden/dept-bre is there a way to ensure we only publish wasm IF the "run tests" workflow succeeds? I.e. depend on the result of another workflow.

The easiest way would be to call the "run tests" workflow in another job and add it to the `needs` key of the publish wasm job. You could probably do some sort of logic to get the run id of the triggered "run tests" workflow, but you would still have to use a loop to constantly check if it completes successfully or not.

File: .github/workflows/build-wasm-internal.yml (hunk 1 above)
Comment (Hinton) [reply .github/workflows/build-wasm-internal.yml:112]: Yes 😞

Comment (vgrassia) [reply .github/workflows/build-wasm-internal.yml:112]: You can do it this way, but you can also use a [workflow_call](https://docs.github.com/en/actions/sharing-automations/reusing-workflows) trigger in the workflow you want to call. There are pros and cons to each method.

**Review by vgrassia — APPROVED** [review]

**Pull request merged**